sda_download/sda_download.go

```diff
@@ -0,0 +1,257 @@
+package sdadownload
+
+import (
+	"encoding/json"
+	"errors"
+	"flag"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"path/filepath"
+	"regexp"
+	"slices"
+	"strings"
+
+	"github.com/NBISweden/sda-cli/helpers"
+	"github.com/vbauerster/mpb/v8"
+	"github.com/vbauerster/mpb/v8/decor"
+)
+
+// Help text and command line flags.
+
+// Usage text that will be displayed as command line help text when using the
+// `help download` command
+var Usage = `
+USAGE: %s sda-download -config <s3config-file> -dataset <datasetID> -url <uri> (-outdir <dir>) [filename(s)]
+
+sda-download:
+	Downloads files from the Sensitive Data Archive (SDA) by using APIs. The user
+	must have been granted access to the datasets (visas) that are to be downloaded.
+	The files will be downloaded in the current directory, if outdir is not defined.
+`
+
+// ArgHelp is the suffix text that will be displayed after the argument list in
+// the module help
+var ArgHelp = `
+	[dataset]
+		The ID of the dataset that the file is part of.
+	[uri]
+		All flagless arguments will be used as sda-download uri.
+	[filename(s)]
+		The name of the file to download.`
+
+// Args is a flagset that needs to be exported so that it can be written to the
+// main program help
+var Args = flag.NewFlagSet("sda-download", flag.ExitOnError)
+
+var configPath = Args.String("config", "", "S3 config file to use for downloading.")
+
+var datasetID = Args.String("dataset", "", "Dataset ID for the file to download")
+
+var url = Args.String("url", "", "The name of the file to download")
```

Comment:
```suggestion
var URL = Args.String("url", "", "The name of the file to download")
```

Don't user variable names that conflict with standard libraries

Reply:
Right